sotodlib/hwp/hwp.py

```diff
@@ -911,3 +911,121 @@ def get_tau_hwp(
+
+def get_hwpss_subscan(
+    aman, signal=None, hwp_angle=None, bin_signal=True, bins=360,
+    lin_reg=True, modes=[1, 2, 3, 4, 5, 6, 7, 8], apply_prefilt=True,
+    prefilt_cfg=None, prefilt_detrend='linear', flags=None,
+    apodize_edges=True, apodize_edges_samps=1600,
+    apodize_flags=True, apodize_flags_samps=200,
+    hwpss_model_name='hwpss_model',replace_model = False, n_buf = 150, window='hann'
+):
+    '''
+    Function to fit and subtract HWPSS by subscan and merge hwpss_model.
+    This assumes that turnaround flags and subscan flags have already been calculated and wrapped.
+    Arguments: same as `get_hwpss`, followed by:
+    
+    replace_model: bool, set True if want to overwrite the hwpss model name in aman (if the same as the hwpss_model_name argument)
+    n_buf: int,  number of samples used to patch the hwpss signal fitted coefficients at the center of each turnaround region.
+    window: str,  scipy.signal.windows.get_window input for the subscan hwpss stitching function. The stitching is done as:
+
+    Take c(s)_i to be the best-fit coefficient for the 4th hwpss harmonic of detector i at sample s:
+
+    At the boundary of subscan_1 and subscan_2 (with constant coefficients c_s1, c_s2, respectively):
+    c(t)_i = {c_s1 for s < s_pivot - n_buf
+              c_s2 + (c_s1 - c_s2)*wn(s) for  s_pivot - n_buf <= s <= s_pivot + n_buf
+              c_s2 for s > s_pivot + n_buf}
+
+    Here, wn(s) is the window function chosen to span 2*n_buf for a smooth transition between coefficients, 
+    and s_pivot is the single sample transition point between subscan_1 and subscan_2.
+    '''
+    if f'{hwpss_model_name}' in aman._assignments:
+        if not replace_model:
+            raise ValueError('hwpss_model_name must be different than what is already in aman if replace_model=False')
+        else:
+            aman.move(hwpss_model_name,None)
+    
+    aman.wrap(hwpss_model_name, np.zeros_like(aman.signal,dtype='float32'),[(0, 'dets'), (1, 'samps')])
+
+    #re-calculate subscan flags with turnarounds included, for apodization purposes
+    subscan_info = tod_ops.flags.get_subscans(aman, flags=None, merge=False, include_turnarounds=True, overwrite=False)
+    num_subscans = len(subscan_info.subscan_flags)
+
+    #transition_flag = np.zeros_like(aman.hwpss_model)
+    coeffs = np.zeros(shape=(aman.dets.count,2*len(modes),len(subscan_info.subscan_flags))) #list of hwpss_subscan harmonic coeff
+    transition_indices = np.array([])
+
+    for j in range(num_subscans):
+        subscan_flag = subscan_info.subscan_flags[j]
+        i = subscan_flag.ranges()[0]
+        sub_aman = aman.restrict('samps', (i[0] + aman.samps.offset, i[1] + aman.samps.offset), in_place=False)
+        sub_aman.move(hwpss_model_name, None)
+        if flags is None:
+            sub_flag = RangesMatrix.from_mask(np.zeros(aman.samps.count, dtype=int)[i[0]:i[1]])
+        else:
+            sub_flag = RangesMatrix.from_mask(flags.mask()[:, i[0]:i[1]])
```

Comment:
This is super inefficient to expand the RangesMatrix out to boolean array and then back into RangesMatrix. What if you just always passed the same full AxisManager but combined the RangesMatrix for the subscan samples with the existing flags i.e. pass to `get_hwpss` the arg: `flags = subscan_flag + flags` and set `merge_model` and `merge_stats` to False so that they're returned and then wrap those into aman as you do on 990-991